libvcx/src/api_lib/api_handle/out_of_band.rs

```diff
@@ -0,0 +1,139 @@
+use std::collections::HashMap;
+
+use crate::aries_vcx::handlers::out_of_band::{OutOfBand, GoalCode};
+use crate::aries_vcx::messages::connection::service::{ServiceResolvable, FullService};
+use crate::aries_vcx::messages::a2a::A2AMessage;
+use crate::api_lib::api_handle::object_cache::ObjectCache;
+use crate::api_lib::api_handle::connection::CONNECTION_MAP;
+use crate::error::prelude::*;
+
+lazy_static! {
+    pub static ref OUT_OF_BAND_MAP: ObjectCache<OutOfBand> = ObjectCache::<OutOfBand>::new("out-of-band-cache");
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct OOBConfig {
+    pub label: Option<String>,
+    pub goal_code: Option<GoalCode>,
+    pub goal: Option<String>,
+}
+
+pub fn is_valid_handle(handle: u32) -> bool {
+    OUT_OF_BAND_MAP.has_handle(handle)
+}
+
+fn store_out_of_band(oob: OutOfBand) -> VcxResult<u32> {
+    OUT_OF_BAND_MAP.add(oob)
+        .or(Err(VcxError::from(VcxErrorKind::CreateOutOfBand)))
+}
+
+pub fn create_out_of_band(config: &str) -> VcxResult<u32> {
```

Comment:
Whould be nicer for config argument to be `OOBConfig`. That way looking at C method `vcx_out_of_band_create` it would also be easier to find out what its `config: *const c_char` is.

Reply:
This might be analogically applicable to other methods too. Move parsing string a layer up